core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -513,7 +513,7 @@ package object config {
         "is written in unsafe shuffle writer. In KiB unless otherwise specified.")
       .bytesConf(ByteUnit.KiB)
       .checkValue(v => v > 0 && v <= ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH / 1024,
-        s"The buffer size must be greater than 0 and less than" +
+        s"The buffer size must be positive and not greater than" +
```

Comment:
not greater than -> less than or equal to

Reply:
Yeah, I could go either way on those. I wouldn't mind standardizing on "less than or equal to", sure. @10110346 would you mind taking one more pass accordingly?